src/decode.rs

```diff
@@ -4900,18 +4895,12 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
 
     // ref_mvs
     if frame_hdr.frame_type.is_inter_or_switch() || frame_hdr.allow_intrabc {
-        f.mvs_ref = rav1d_ref_create_using_pool(
-            c.refmvs_pool,
-            ::core::mem::size_of::<refmvs_temporal_block>()
-                * f.sb128h as usize
-                * 16
-                * (f.b4_stride >> 1) as usize,
+        // TODO fallible allocation
+        f.mvs = Some(
+            (0..f.sb128h as usize * 16 * (f.b4_stride >> 1) as usize)
+                .map(|_| Default::default())
+                .collect(),
         );
```

Comment:
This took way too long to debug, so annoying.  I forgot the `Some` here, which may the `.collect()` and `Default::default()` use `Option`'s `impl FromIterator` instead of into an `Arc<[_]>`.